java/src/jmri/util/NonNullArrayList.java

```diff
@@ -1,41 +1,44 @@
 package jmri.util;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Objects;
+
 import javax.annotation.Nonnull;
 
 /**
  * An ArrayList that SpotBugs understands will never contain null elements.
  *
  * @see java.util.ArrayList
  * @see java.util.List
- * @author	Bob Jacobsen, Copyright (C) 2017
+ * @author Bob Jacobsen, Copyright (C) 2017
+ * @param <E> the class of object in the List
  */
 public class NonNullArrayList<E> extends ArrayList<E> {
 
     @Override
-    public boolean add(@Nonnull E e) {
-        if (e == null) throw new IllegalArgumentException("NonNullArrayList.addAll cannot add null item");
+    public boolean add(E e) {
+        Objects.requireNonNull(e, "NonNullArrayList.addAll cannot add null item");
         return super.add(e);
     }
 
     @Override
-    public void add(int i, @Nonnull E e) {
-        if (e == null) throw new IllegalArgumentException("NonNullArrayList.addAll cannot add null item");
+    public void add(int i, E e) {
+        Objects.requireNonNull(e, "NonNullArrayList.addAll cannot add null item");
         super.add(i, e);
     }
 
     @Override
-    public boolean addAll(Collection<? extends E> c) { // ideally, would be "extends @Nonnull e", but that's not this annotation
+    public boolean addAll(Collection<? extends E> c) {
         for (E e : c ) {
-            if (e == null) throw new IllegalArgumentException("NonNullArrayList.addAll cannot accept collection containing null");
+            Objects.requireNonNull(e, "NonNullArrayList.addAll cannot accept collection containing null");
         }
         return super.addAll(c);
     }
 
     @Override
-    public boolean addAll(int i, Collection<? extends E> c) { // ideally, would be "extends @Nonnull e", but that's not this annotation
```

Comment:
I'd prefer the `ideally, would be "extends @Nonnull e"` comment stay, either inline or as on the Javadoc `@param`.  (several places). Having that in comments, as opposed to figuring it out from the code, seems useful.

Reply:
I'll add a documentation comment.